sunpy/map/mapbase.py

```diff
@@ -572,6 +572,27 @@ def max(self, *args, **kwargs):
         """
         return self.data.max(*args, **kwargs)
 
+    @property
+    def unit(self):
+        """
+        Unit of the map data.
+
+        This is taken from the 'BUNIT' FITS keyword. If no 'BUNIT' entry is
+        present in the metadata then this returns `None`. If the 'BUNIT' value
+        cannot be parsed into a unit a warning is raised, and `None` returned.
+        """
+        unit_str = self.meta.get('bunit', None)
+        if unit_str is None:
+            return
+        else:
+            unit = u.Unit(unit_str, parse_strict='warn')
+            if isinstance(unit, u.UnrecognizedUnit):
+                warnings.warn(f'Could not parse unit string "{unit_str}". '
+                              'See the above warning for more information',
```

Comment:
Would it not be cleaner to capture and parse the astropy warning, it's pretty long and mostly irrelevant for the map user here (i.e. it suggests things they might mean, which isn't really how you interact with a FITS header?)

Reply:
👍 maybe a link to the FITS unit standard would be good in this warning so a user can go and look up and alternative to fix the metadata

Reply:
How about now?